watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Activities/LandingPageActivity.java

```diff
@@ -190,6 +223,32 @@ public void onCompleted(GraphUser user, com.facebook.Response response) {
         }
     }
 
+    /**
+     * Registers the application with GCM servers asynchronously.
+     * <p>
+     * Stores the registration ID and app versionCode in the application's
+     * shared preferences.
+     */
+    public void registerInBackground(final HashMap<String, Object> objectParams, final String type) {
+        new AsyncTask<String, String, String>() {
+            @Override
+            protected String doInBackground(String... params) {
+                String msg = "";
+                try { objectParams.put("registration_id", mGcm.register(SENDER_ID)); }
+                catch (IOException ex) { msg = "Error :" + ex.getMessage(); }
```

Comment:
Can we just log the error from inside this method? Then we would just have <code>AsyncTask()</code>